api-report/azure-client.api.md

```diff
@@ -26,26 +24,28 @@ export class AzureAudience extends ServiceAudience<AzureMember> implements IAzur
 
 // @public
 export class AzureClient {
-    constructor(connectionConfig: AzureConnectionConfig, logger?: ITelemetryBaseLogger | undefined);
-    createContainer(containerSchema: ContainerSchema): Promise<{
-        container: FluidContainer;
+    constructor(props: AzureClientProps);
+    createContainer<T extends LoadableObjectClassRecord>(containerSchema: ContainerSchema<T>): Promise<{
+        container: FluidContainer<T>;
         services: AzureContainerServices;
     }>;
-    getContainer(id: string, containerSchema: ContainerSchema): Promise<{
-        container: FluidContainer;
+    getContainer<T extends LoadableObjectClassRecord>(id: string, containerSchema: ContainerSchema<T>): Promise<{
```

Comment:
just a bit of feedback. bohemia finds the existing generics on DataObject far to complex to use. this is a common problem with generics. you should be really cautious with their usage and ensure the usage remains ergonomic

Reply:
Yea, this is my biggest concern. The problem being that when you try to read the generics it's confusing but when you try to use it is helpful. It's a tough balance because I really don't like casts.